### ISSUE TITLE
False positive with Container::get not nullable

### ISSUE THREAD

**Issue by noemi-salaun** [phpstan/phpstan-symfony#74]
`Symfony\Component\DependencyInjection\ContainerInterface::get()` can return null when using `ContainerInterface::NULL_ON_INVALID_REFERENCE`

```php
$stack = $container->get('request_stack', ContainerInterface::NULL_ON_INVALID_REFERENCE);
if (null === $stack) {
    return null;
}
```

> Strict comparison using === between null and Symfony\Component\HttpFoundation\RequestStack will always evaluate to false.

**Comment by lookyman** [comment]
I am actually thinking about straight up disallowing the get on containers. What are your thoughts about that?

Otherwise, this one might currrently be tough to fix. I have some ideas, but it would possibly require some major refactoring, and I am not sure about the results. Plus I am currently swamped with other work.

So for now, please just mark the error as ignored, and I promise to look at it as soon as I have some spare time. 😊 Thanks.

**Comment by noemi-salaun** [comment]
> I am actually thinking about straight up disallowing the get on containers. What are your thoughts about that?

I think it's a little too opinionated to simply disallow its usage. Maybe in a package like `phpstan/phpstan-strict-rules` but not as default in this package.

**Comment by ostrolucky** [comment]
There is an opposite problem too. In case containerXmlPath is not specified, it will claim return type is nullable. We have these kind of tests a lot

```php
class FooTest extends \Symfony\Bundle\FrameworkBundle\Test\KernelTestCase
{
    private Foo $foo;

    protected function setUp(): void
    {
        $this->foo = $this->getContainer()->get(Foo::class);
    }
```

but phpstan complains here with
> Property FooTest::$foo (Foo) does not accept object|null.

**Comment by borislt** [comment]
@ostrolucky I stumble upon the same error in my test, how do you handle it?
